org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradleProjectImporter.java

```diff
@@ -629,10 +636,16 @@ private static File getGradleInitScript(String scriptPath) {
 	 * Create a temp file as the Gradle init script.
 	 */
 	private static File getGradleInitScriptTempFile(String scriptPath) {
+		File cachedFile = initScriptFileCache.get(scriptPath);
```

Comment:
A different approach I was thinking of is that we could create a fixed temporary file for an identical scriptPath. For example, you could calculate the md5 value of the scriptPath and create a file in the tmp directory using md5 as the filename. This way you avoid creating many temporary files for the init scripts, and you don't need to delete them as well.